### PR TITLE
[doc] Update utilities.rst

### PR DESCRIPTION
**Pull request by yuanming-hu**
Related issue = #941 

 - Moved `gdb` related paragraph from `contributor_guide.rst` to `utilities.rst`
 - `utilities.rst` is now titled `Developer utilities` and made part of `Contributing`.

I left `Documentation` in `contributor_guide` since that's essential to anyone working on the documentation.

[[Click here for the format server]](http://kun.csail.mit.edu:31415/)

### CODE REVIEW

File: docs/utilities.rst [hunk 1]
```diff
@@ -23,7 +23,7 @@ You can also override default logging level by setting the environment variable
 
 .. _regress:
 
-Benchmarking and Regression Tests
+Benchmarking and regression tests
```
Comment (archibate) [inline docs/utilities.rst:26]: Cool! Could you also move `Tips on the Taichi compiler development`, `ti test`, `ti doc` to here and only left a **link** in contribute.rst? I think they are developer utilities, not contributor general guide.

File: docs/contributor_guide.rst [hunk 2]
```diff
@@ -137,7 +137,7 @@ Existing tags:
 - ``[Metal], [OpenGL], [CPU], [CUDA], [AMDGPU], [LLVM]``: backends;
 - ``[LLVM]``: the LLVM backend shared by CPUs and CUDA;
 - ``[Lang]``: frontend language features, including syntax sugars;
-- ``[Std]``: standard library, e.g. `ti.Matrix` and `ti.Vector`;
+- ``[Std]``: standard library, e.g. ``ti.Matrix`` and ``ti.Vector``;
```
Comment (archibate) [inline docs/contributor_guide.rst:140]: Thank for adding this, btw, could you also add a `[Benchmark]` at some point? See #981.

File: docs/utilities.rst (hunk 1 above)
Comment (yuanming-hu) [reply docs/utilities.rst:26]: As mentioned above I think those are essential tools for all developers so I left them there.

File: docs/contributor_guide.rst (hunk 2 above)
Comment (yuanming-hu) [reply docs/contributor_guide.rst:140]: Please feel free to do that on your own in a separate PR. Thanks.

Comment (archibate) [reply docs/contributor_guide.rst:140]: Added in #981 now!

Comment (yuanming-hu) [reply docs/contributor_guide.rst:140]: Thank you!

**Pull request merged**